README.md

```diff
@@ -1,101 +1,106 @@
-3. When prompted, authenticate with your Microsoft account and consent to the permissions that the application needs.
-4. Follow the prompts to send a message from your account to yourself or someone else.
-   
-## Questions and comments
-
-We'd love to get your feedback about the Microsoft Graph API Console App. You can send your questions and suggestions in the [Issues](https://github.com/microsoftgraph/console-csharp-connect-sample/issues) section of this repository.
-
-Questions about Microsoft Graph development in general should be posted to [Stack Overflow](https://stackoverflow.com/questions/tagged/microsoftgraph). Make sure that your questions or comments are tagged with [microsoftgraph].
-
-## Contributing ##
-
-If you'd like to contribute to this sample, see [CONTRIBUTING.MD](/CONTRIBUTING.md).
-
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
-  
-## Additional resources
-
-- [Other Microsoft Graph Connect samples](https://github.com/MicrosoftGraph?utf8=%E2%9C%93&query=-Connect)
-- [Microsoft Graph](https://developer.microsoft.com/en-us/graph)
-
-## Copyright
-Copyright (c) 2019 Microsoft. All rights reserved.
+# Microsoft Graph C# Console Connect Sample
+
+## Table of contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Register the application](#Register-the-application)
+* [Build and run the sample](#build-and-run-the-sample)
+* [Cloning or downloading this repository](#cloning-or-downloading-repo)
+* [Configure your Azure AD tenant](#configuring-Azure-AD-tenant )
+* [Configure the sample to use your Azure AD tenant](#configuring-sample-to-use-Azure-AD-tenant)
+* [Build and run the sample](#build-and-run-sample)
+* [Questions and comments](#questions-and-comments)
+* [Contributing](#contributing)
+* [Additional resources](#additional-resources)
+
+## Introduction
+
+This sample shows how to connect a Windows console application to a Microsoft work or school (Azure Active Directory) or personal (Microsoft) account using the Microsoft Graph API. It uses the Microsoft Graph API to retrieve a user's profile picture, upload the picture to OneDrive, create a sharing link, and send an email that contains the photo as an attachment and the sharing link in its text. It uses the Microsoft Graph .NET Client Library to work with data returned by Microsoft Graph. The sample uses the Azure AD v2.0 endpoint, which enables users to sign in with either their personal or work or school Microsoft accounts.
+
+The sample uses the Microsoft Authentication Library (MSAL) for authentication.
+
+## Prerequisites
+
+This sample requires the following:
+- [Visual Studio](https://www.visualstudio.com/en-us/downloads) 

```

Comment:
Do we really need this? There is already a link to Visual Studio below (line 27).